### PR TITLE
Hostname value standard compliance fix

### LINKED ISSUE #11134: Document usage of hostname has to comply with standards
### Describe the bug

Follow-up of #10930 (see latest comment) 

At the moment, we're not checking the value for `hostname` for validity. These checks are also not done by quarkus itself. Validity (as of RFC 952/1123 etc., see https://en.wikipedia.org/wiki/Hostname#Syntax) is needed for keycloak to work correctly. 

### Version

current

### Expected behavior

~~all hostnames are converted to lowercase first, and then checked for general validity. If valid, they should be used, if not, an error should be shown.~~

"how to" for adding the hostname value is added to hostname guide ( see #11139 for details on changing)
### Actual behavior

Keycloak/Quarkus just accepts invalid hostname values like `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa` (more than 253 chars).
 

### How to Reproduce?

Run e.g. `./kc.sh start --hostname=LOCALHOST --https-certificate-file ../conf/id.keycloak.test.pem --https-certificate-key-file ../conf/id.keycloak.test-key.pem`, create an initial admin user and login to admin console. The admin console will not show up apart from the header menu (see e.g. #10930)  , as the hostname is not standard compliant.

### Anything else?

-

### PR DESCRIPTION
**Pull request by DGuhr**
Closes #11134

### CODE REVIEW

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnamePropertyMappers.java [hunk 1]
```diff
@@ -1,6 +1,14 @@
 package org.keycloak.quarkus.runtime.configuration.mappers;
 
 
+import com.google.common.net.InternetDomainName;
```
Comment (pedroigor) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnamePropertyMappers.java:4]: I would avoid using Guava directly. Although it is available at runtime, we should avoid if possible a compile-time dependency.

In the worst case, you should try using something from Apache libs as we are using it anyways.

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnamePropertyMappers.java [hunk 2]
```diff
@@ -44,6 +53,17 @@ private HostnamePropertyMappers(){}
         };
     }
 
+    private static String getValidatedLowercaseHostname(String value, ConfigSourceInterceptorContext configSourceInterceptorContext) {
+
+        String hostnameVal = value.toLowerCase(Locale.ROOT);
```
Comment (pedroigor) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnamePropertyMappers.java:58]: Are you sure the `Local.ROOT` local is the best here? IIRC, there are some languages that might not lower case some letters.

Why not let it `user.language` dependent ?

**Review by pedroigor — CHANGES_REQUESTED** [review]
I wouldn't add this validation as is but perhaps if it means using some more strict validation for security purposes (e.g.: something recommended by OWASP).

Even in this case, I'm unsure how far this can break people so perhaps we should also have some flag to skip the validation?

**Comment by pedroigor** [comment]
Btw, we have other places that perhaps deserve this? For instance, `db-url-host` ?

Also, is it still possible to use an IP address?

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnamePropertyMappers.java (hunk 2 above)
Comment (DGuhr) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnamePropertyMappers.java:58]: hum you are right. I think we should go with Locale.English here, that should make sure it always gets lowercased correctly (or are there domain names not using a-z but another alphabet? i don't think so, but not 100% sure). Will fix :)

Comment (DGuhr) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnamePropertyMappers.java]: so.. should we remove the actual validation check for now and just go with the lowercase approach to mitigate the actual issue? perhaps i overdid it a bit, wasnt aware of all the edge cases

Comment (pedroigor) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnamePropertyMappers.java]: Hope now you see the challenge around hostname :)

For me, I would just give up on this change and update the documentation accordingly with a note/warning/tip.

Comment (DGuhr) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnamePropertyMappers.java]: sounds like a plan. closing this for now and will create another PR with updated hostname guide.

Comment (DGuhr) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnamePropertyMappers.java]: https://github.com/keycloak/keycloak/pull/11162

**Pull request closed without merging**